src/scripts/domParsers/khanAcademyVideoExtractor.ts

```diff
@@ -0,0 +1,50 @@
+import {VideoExtractor} from "./videoExtractor";
+import {VideoUtils} from "./videoUtils";
+import {YoutubeVideoExtractor} from "./youtubeVideoExtractor";
+
+import {Utils} from "../utils";
+
+export class KhanAcademyVideoExtractor implements VideoExtractor {
+	private youtubeExtractor: YoutubeVideoExtractor;
```

Comment:
Personally I prefer not using the extends and all those overrides because typescript currently doesn't support amazing support for safely overriding (i.e., it's easy for the dev to make a mistake such as forgetting to override something, or mispelling a method name). The thing I'd do would be to add a comment here saying that we use the same method as youtube to create embedded videos.